.github/workflows/e2e.yml

```diff
@@ -0,0 +1,140 @@
+name: E2E Suite
+
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - ".github/workflows/e2e.yml"
+      - "lib/**"
+      - "proto/**"
+      - "scripts/**"
+      - "skaffold/**"
+      - "src/**"
+      - "tests/**"
+      - "distribution/**"
+      - "Cargo.lock"
+      - "Cargo.toml"
+      - "build.rs"
+      - "docker-compose.yml"
+      - "Makefile"
+      - "rust-toolchain"
+  pull_request: {}
```

Comment:
If I recall, the paths must also be specified on the `pull_request` block as well. Otherwise these will run on every pull request.

Reply:
I'll double-check. If it's the case - the `tests.yaml` has to be corrected - it only has those specified for `push`, exactly like I have here.

Reply:
Good catch! I've created https://github.com/timberio/vector/pull/3366